app/src/main/java/unipd/se18/ocrcamera/AllergenListAdapter.java

```diff
@@ -0,0 +1,103 @@
+        mResource = resource;
+        //set an AllergenManager used to modify users' allergens list
+        mAllergensManager= new AllergensManager(mContext);
+        selectedAllergens = mAllergensManager.getSelectedAllergensList();
+    }
+
+    /**
+     * this method deals with the adaptation of the single allergen view
+     * to the ListView of AllergensSearchFragment
+     * @param position of the Allergen
+     * @param convertView the view to be adapted
+     * @param parent viewGroup
+     * @return convertView the adapted View
+     */
+    @NonNull
+    @Override
+    public View getView(final int position, View convertView, ViewGroup parent) {
+        try {
+            //Allergen information
+            final Allergen mAllergen= getItem(position);
+            String name = mAllergen.getCommonName();
+            boolean selected = selectedAllergens.contains(mAllergen);
+
+            //create the Allergen in the layout
+            LayoutInflater inflater = LayoutInflater.from(mContext);
+            convertView = inflater.inflate(mResource, parent, false);
+
+            //set TextView to name of the Allergen
+            Log.i(TAG, "Set allergen name to "+name);
+            TextView allergenTv = convertView.findViewById(R.id.singleAllergenTv);
+            allergenTv.setText(name);
+
+            //set ToggleButton to checked or not depending on the value of "selected"
+            ToggleButton allergenToggleButton = convertView.findViewById(R.id.toggleButton);
+            Log.i(TAG, "Set button state to "+selected);
+            allergenToggleButton.setChecked(selected);
+
+            //change allergen selection state and remove from/add to user's list if the button is clicked
+            allergenToggleButton.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+                @Override
+                public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
+                    //check if it is still selected or not (user may press the button multiple times
+                    boolean isSelected= selectedAllergens.contains(mAllergen);
+                    //add or remove the one chosen by the user
+                    if (isSelected){
+                        Log.i(TAG, "remove allergen from user's list");
+                        selectedAllergens.remove(mAllergen);
+                    }
+                    else {
+                        Log.i(TAG, "add allergen to user's list");
+                        selectedAllergens.add(mAllergen);
+                    }
+                    Log.i(TAG, "Lista degli allergeni: "+selectedAllergens.toString());
+                    mAllergensManager.updateSelectedAllergens(selectedAllergens);
+                }
+            });
+        }
+        catch (NullPointerException npe){
+            Log.e(TAG, "Missing Allergens' parameters");
+            npe.printStackTrace();
+        }
```

Comment:
Is there any better way to manage this exception ? What happens if the exception happens ?

Reply:
As i said to the other reviewer, i forgot to write in the pull request summary that since I am still inexperienced I didn't really know how to handle this NullPointerException, but I did know this was probably not the way to do it. I thought about giving a message to the user or even closing the Activity, but both seemed too User-unfriendly to me because there is nothing he can do to solve the problem.